### DISCUSSION TITLE
Expected time of release of the beta version

### DISCUSSION THREAD

**Discussion by heEXDe** [NiceNode/nice-node#543] [General]
Hello.
A simple (perhaps seemingly) question:
When can we probably expect the project to move from alpha to beta and release the first beta version, ready to be downloaded and installed? 
(by the way) Thanks for your work. It's an interesting project.

**Comment by jgresham** [comment]
Hey @heEXDe, thanks for the question. Some issues that are keeping me from moving it to beta:

- Certain node config changes or updates require users to remove a node and re-add the node in NiceNode. This is due to NiceNode not automatically updating node config seamlessly and I consider this a known, critical bug
- On Mac and Windows, I need to build an automated test environment. There have been intermittent issues that arise when running nodes for long period of time which cause them to get behind on syncing, and the Windows install process has a lot of moving parts (it relies on installing WSL2)
- Even if I can get it to Beta (for Linux) in the next month, there are a few big features that I would like to add which I will likely delay the Beta in order to implement those features first (Permissionless node library/options is one)

Generally how I view stages:
alpha: known or very likely bugs that will cause users to reinstall or mostly reconfigure the app
beta: only known bugs are minor, but there may be many. widely untested. however, bugs are not causing issues with the core functionality of the app
production: you can count on it for its core functionality and future updates are not expected the break the app

**Reply by heEXDe** [reply]
Thanks for your answer (and a comprehensive one at that).
It's probably best to approach big projects in small steps - lots of small steps, rather than one jump into deep water. This helps to 'not drown' in the 'depths' of the project.
Plus: the most satisfaction comes from working on difficult challenges.
Thanks again for the work you put in. Easier node running means decentralization at a basic and therefore crucial level.
